generator/src/main/java/com/scottlogic/deg/generator/generation/string/factorys/RandomStringFactory.java

```diff
@@ -22,18 +22,29 @@
 import dk.brics.automaton.Transition;
 
 import java.util.List;
+import java.util.OptionalInt;
 
 public class RandomStringFactory {
 
+    public String createRandomString(State state, RandomNumberGenerator random) {
+        return createRandomString("", OptionalInt.empty(), state, random);
+    }
+
     public String createRandomString(
             String currentString,
+            OptionalInt shortestStringLength,
             State state,
-            int minLength,
-            int maxLength,
             RandomNumberGenerator random) {
 
-        if (finishCreating(currentString, state, minLength, maxLength, random)) {
-            return currentString;
+        OptionalInt newShortest = shortestString(shortestStringLength, currentString, state);
+
+        if (finishCreating(state)) {
+            int max = currentString.length();
+            int min = newShortest.getAsInt();
+            int diff = max - min;
+            int randomDiff = random.nextInt(diff + 1);
+            int length = min + randomDiff;
+            return currentString.substring(0, length);
```

Comment:
Note to self: taking the substring could violate the regex. We can fix this by storing all of the valid intermediary strings, and choosing randomly from them at the end of generation.